packages/core/src/internal/temporal-util.ts

```diff
@@ -428,13 +428,20 @@ export function assertValidNanosecond (
   )
 }
 
-export function assertValidZoneId (fieldName: string, zoneId: string): void {
+const timeZoneValidityCache = new Map();
+
+export function assertValidZoneId(fieldName: string, zoneId: string): void {
+  let result = false;
+
   try {
-    Intl.DateTimeFormat(undefined, { timeZone: zoneId })
+    if (timeZoneValidityCache.get(zoneId)) return;
```

Comment:
The linter is falling because of `packages/core/src/internal/temporal-util.ts:437:9: Unexpected any value in conditional. An explicit comparison or type cast is required. (@typescript-eslint/strict-boolean-expressions)`

```suggestion
    if (timeZoneValidityCache.get(zoneId) === true) return;
```

Reply:
It also worth to throw if the cached === false.

Reply:
Done